File: civitasRuby/lib/civitas/sorpresa_por_casa_hotel.rb
#encoding:utf-8
=begin
Authors: Esther García Gallego
         Yesenia González Dávila
         Grupo B3
=end

module Civitas

class SorpresaPorCasaHotel < Sorpresa
  
  def initialize (valor, texto)
    super("*Por casas y hoteles*", nil, valor, texto, nil)
  end
  
  def aplicarAJugador (actual, todos)
    if jugadorCorrecto(actual, todos)
      informe(actual, todos)
      todos[actual].modificarSaldo(@valor * todos[actual].cantidadCasasHoteles)
    end
  end
end

end

File: P5/civitasJava/src/civitas/SorpresaPorCasaHotel.java

package civitas;

import java.util.ArrayList;

public class SorpresaPorCasaHotel extends Sorpresa {
    
    SorpresaPorCasaHotel (int valor, String texto) {
        super ("*Por casa/hotel*");
        this.texto = texto;
        this.valor = valor;
    }
    
    
    /** Se modifica el saldo del jugador actual con el valor de la sorpresa
     *  multiplicado por el núm de casas y hoteles del jugador. */
    @Override
    void aplicarAJugador (int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)) {
            informe (actual, todos);
            System.out.println(valor * todos.get(actual).cantidadCasasHoteles());
            todos.get(actual).modificarSaldo (valor*todos.get(actual).cantidadCasasHoteles());
        }
    }
    
}
